libafl/src/stages/calibrate.rs

```diff
@@ -161,21 +194,38 @@ where
                 history_map.resize(map_len, O::Entry::default());
             }
 
-            for (first, (cur, history)) in
-                map_first.iter().zip(map.iter().zip(history_map.iter_mut()))
+            for (idx, (first, (cur, history))) in map_first
+                .iter()
+                .zip(map.iter().zip(history_map.iter_mut()))
+                .enumerate()
             {
                 if *first != *cur && *history != O::Entry::max_value() {
                     *history = O::Entry::max_value();
-                    unstable_entries += 1;
+                    unstable_entries.push(idx);
                 };
             }
 
             i += 1;
         }
 
         #[allow(clippy::cast_precision_loss)]
-        if unstable_entries != 0 {
-            *state.stability_mut() = Some((map_len - unstable_entries) as f32 / (map_len as f32));
+        if unstable_entries.len() != 0 {
+            // If we see new stable entries executing this new corpus entries, then merge with the existing one
+            if !state.has_metadata::<UnstableEntriesMetadata>() {
+                state.add_metadata::<UnstableEntriesMetadata>(UnstableEntriesMetadata::new(
+                    HashSet::from_iter(unstable_entries),
+                    map_len,
+                ))
+            } else {
+                let existing = state
+                    .metadata_mut()
+                    .get_mut::<UnstableEntriesMetadata>()
+                    .unwrap();
+                for item in unstable_entries {
+                    existing.unstable_entries.insert(item); // Insert newly found items
```

Comment:
same here? maybe I do no fully understand the code :)

Reply:
existing.unstable_entries is a HashSet()
if there's an entry that is already in existing.unstable_entries, then nothing happens (because it is a Set)